scripts/cell_simulation.py

```diff
@@ -106,58 +109,61 @@ def create_parser() -> argparse.Namespace:
     return args
 
 
-def compose_save_name(params: argparse.Namespace, *, tree_no: int) -> str:
-    """Composes save name for the results."""
-    save_name = (
-        f"seed_{params.seed}_"
-        f"n_trees_{params.n_trees}_"
-        f"n_cells_{params.n_cells}_"
-        f"n_mutations_{params.n_mutations}_"
-        f"fpr_{params.fpr}_"
-        f"fnr_{params.fnr}_"
-        f"na_rate_{params.na_rate}_"
-        f"observe_homozygous_{params.observe_homozygous}_"
-        f"strategy_{params.strategy}"
-    )
-    if tree_no is not None:
-        save_name += f"_tree_{tree_no}"
-    return save_name
-
-
-def gen_sim_data(
-    params: argparse.Namespace,
-    rng: PRNGKeyArray,
-    *,
-    tree_no: int,
-) -> None:
-    """
-    Generates cell mutation matrix for one tree and writes to file.
+def run_sim(params: argparse.Namespace) -> None:
+    """Generate {n_trees} of simulated data and save to disk.
 
     Args:
-        params: input parameters from parser
+        params: argparse.Namespace
             input parameters from parser for simulation
-        rng: JAX random number generator
-        tree_no: int - optional
-            tree number if a series is generated
+
     Returns:
         None
     """
-    ############################################################################
-    # Parameters
-    ############################################################################
+
+    ###############################
+    # Get Tree information from TreeId
+    ###############################
+    # make tree id from tree path
+    true_tree_fp = params.true_tree_fp
+    # Get the filename without the file extension and directories
+    tt_filename = true_tree_fp.name
+    tt_filename_without_extension = tt_filename.split(".")[0]
+
+    tree_id = TreeId.from_str(tt_filename_without_extension)
+
+    ###############################
+    # Get Cell Simulation Id and set up CellSimulationModel
+    ###############################
+    # make cell simulation id
+    cell_sim_id = CellSimulationId(
+        seed=params.seed,
+        tree_id=tree_id,
+        n_cells=params.n_cells,
+        strategy=params.strategy,
+        fpr=params.fpr,
+        fnr=params.fnr,
+        na_rate=params.na_rate,
+        observe_homozygous=params.observe_homozygous,
+    )
+
+    # set up the CellSimulationModel from the params and Tree Information from TreeId
     params_dict = vars(params)
+    params_dict["n_mutations"] = tree_id.n_nodes - 1
```

Comment:
Note how the `n_mutations` parameter is now defined by the true trees filename i.e. `TreeId`.  Just one wildcard in the snakemake to rule both

Reply:
Mmhh - alright - I think I still don't quite understand your method -  how to expand the dictionary `scenario` way to so many different parameters without having very long dictionaries.